### ISSUE TITLE
Recommend enabling the `-fPIC` compiler flag by default

### ISSUE THREAD

**Issue by micahsnyder** [json-c/json-c#508]
I would like to propose enabling the `-fPIC` CFLAGS / CXXFLAGS option by default in the build system.

I'll try to explain why. Json-c appears to have some of the same symbol names in common with the jansson, json-glib, and perhaps others.  See https://gitlab.gnome.org/GNOME/json-glib/issues/33 for details.  

An application which depends on a shared library that in turn depends on libjson-c may experience occasional crashes if they require a different JSON library for their own needs.  

The specific issue that I'm facing is that ClamAV's libclamav.so library must link statically with libjson-c.a in order to mitigate crashes in applications that depend on libclamav.so.  However, to link a shared object with a static library, the static library must be built with `-fPIC` or else you'll get a build failure like this:

```
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu/libjson-c.a(linkhash.o): relocation R_X86_64_PC32 against symbol `stdout@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
```

I can't make static linking with libjson-c default in libclamav.so, unless the `-fPIC` option is default, because it will fail to compile for most people.  I would, however, prefer if Linux distributions do not provide a libclamav.so to their users which may cause occasional crashes.

**Comment by micahsnyder** [comment]
Bump.  Ticket https://github.com/json-c/json-c/issues/621 is effectively a duplicate request.  The solution is to compile the libjson-c-static.a library with `-fPIC` so it can be statically linked into other libraries so downstream apps can use whichever json library they prefer without risk of crashing.

**Comment by hawicz** [comment]
Yeah, ok, let's do that, but provide a way to turn that off  (e.g. `cmake -DDISABLE_STATIC_FPIC=on`).  Can you make the changes to CMakeLists.txt and README.md and submit a PR?

**Comment by micahsnyder** [comment]
@hawicz Sure, working on it now.  I have something prepped but am doing some local testing before I submit the PR.

**Comment by hawicz** [comment]
Closing this, since I just merged #634
